source/commands/manage-mods.js

```diff
@@ -0,0 +1,64 @@
+const Command = require('../classes/Command.js');
+const { modRoleId, isModerator, addModerator, removeModerator } = require('../helpers.js');
+
+const options = [];
+const subcomands = [
+	{
+		name: "promote",
+		description: "(moderator) Add a user to the moderator list",
+		optionsInput: [
+			{ type: "User", name: "promotee", description: "The user's mention", required: true, choices: [] },
+		]
+	},
+	{
+		name: "demote",
+		description: "(moderator) Remove a user from the moderator list",
+		optionsInput: [
+			{ type: "User", name: "demotee", description: "The user's mention", required: true, choices: [] }
+		]
+	}
+];
+module.exports = new Command("manage-mods", "(moderator) Promote/demote a user to moderator", true, options, subcomands);
+
+/**
+ * @param {import('discord.js').Interaction} interaction
+ */
+module.exports.execute = (interaction) => {
+	if (interaction.options.getSubcommand() === "promote") {
+		// Add a Moderator: add to list, give role and channel permissions
+		if (isModerator(interaction.user.id) || !interaction.member.manageable) {
```

Comment:
We need to check that the interaction user is a moderator regardless of subcommand, so that should be checked first.